vtm.js

```diff
@@ -116,41 +121,59 @@ app.component('sheet',{
 //STAT SECTION
 app.component('stat-section',{
   props: ['stats'],
-  template: `
+  data() {
+    return { 
+      resourceCount : null,
+  }},
+  template: `<div>{{resourceCount}}</div>
              <div class="statSection">
                <h2>{{stats.display}}</h2>
                <stat-category v-for="list in stats.data"
                :key="list.display"
                :categ="list"
+               :resource="this.resourceCount"
+               :scale="stats.resource.length - 1"
                ></stat-category></div>`,
+  created(){
```

Comment:
pomohol by popis co ta funkcia robi

Reply:
ktora funkcia? v kode nevidim ziadnu funkciu
mozno created ale to je lifecycle hook

Reply:
ok, tak prosim o doplnenie popisu ako funguje lifecycle hook created, co robi, ako to robi

Reply:
https://v3.vuejs.org/api/options-lifecycle-hooks.html#created

Reply:
ano, to co je napisane v tom odkaze chapem, ale nechapem co a preco robi tento kusok kodu:

```
    let tmp = Array(this.stats.resource.length).fill(0);
    for (var i = 0; i < this.stats.data.length; i++) {
      for (var j = 0; j < this.stats.data[i].list.length; j++) {
        tmp[this.stats.data[i].list[j].value]++;
      }
    }
    tmp.forEach(
      (element, index) => (tmp[index] = this.stats.resource[index] - element)
    );
    this.resourceCount = tmp;
```

pacil by sa mi kratky komentar typu

// uvodna hodnota this.resourceCount je [pocet_skillov, obmedzenie1, neviem2, neviem3, neviem8 ]

Reply:
`let tmp = Array(this.stats.resource.length).fill(0);` vytvori pole counterov
prejdi data a`this.stats.data[i].list[j].value` counter na tejto pozicii zvys o 1
tmp.forEach(
      (element, index) => (tmp[index] = this.stats.resource[index] - element)
    );` urob pole resource - tmp`

Reply:
> `let tmp = Array(this.stats.resource.length).fill(0);` vytvori pole counterov
o dlzke this.stats.resource a naplni ho nulami - co je v tejto chvili resource? To su proste ty chybajuce popisy attributov. Ty sa v tom vyznas, ale ked to citam ja, tak to nevidim - asi som blby :-)

Reply:
mohol by som si zapnut debugger a pozerat na to, ale robim review kodu, ktory chcem aby bol citelny

Reply:
no ale chapes moj problem ze v jednom kode si stazujes ze robim `const neco = data.interna_hodnota` a v druhom kode kde to nerobim zas hovoris ze tomu nerozumies

Reply:
no nechapem :-) ako vzdy...

ked mas objekt, tak obvykle v metode toho objektu volas atributy toho objektu this.nieco - this vyjadruje, ze sa to tyka instancie tohoto objektu
ty robis to, ze das `const nie = this.tamto.nieco` a potom pouzivas zkratene `nie` v dalsom kodu, takze ten kto predtym chapal co je to `this.tamto.nieco` si navyse musi pametat, ze teraz sa to vola inac.

Okrem toho ja navyse neviem, co je `this.tamto.nieco`, tak sa na to pytam a predpokladam, ze ked to neviem ja, tak ten dalsi, co to po mne niekedy bude citat, tak to tiez nebude vediet, takze navrhujem, aby sa pridala dokumentacia, ktora by to pripadne osvetlila. Druha vec je, ze to asi nikto dalsi po tebe nebude citat - ale ak mas v umyslu to ukazat potencionalnemu zamestnavatelovi, tak toto je bod, kde hovoris: Viem pisat kod, ktory dokazy pochopit aj ostatny v tymu.